File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/config/ConnectionFactoryInterceptor.java
package io.micronaut.rabbitmq.docs.config;

import com.rabbitmq.client.ConnectionFactory;
import com.rabbitmq.client.impl.DefaultCredentialsProvider;
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import jakarta.inject.Singleton;

@Singleton
public class ConnectionFactoryInterceptor implements BeanCreatedEventListener<ConnectionFactory> {

    @Override
    public ConnectionFactory onCreated(BeanCreatedEvent<ConnectionFactory> event) {
        ConnectionFactory connectionFactory = event.getBean();
        connectionFactory.setCredentialsProvider(new DefaultCredentialsProvider("guest", "guest"));
        return connectionFactory;
    }
}

File: docs-examples/example-python/src/test/python/micronaut/rabbitmq/docs/config/ConnectionFactoryInterceptor.py
from com.rabbitmq.client import ConnectionFactory
from com.rabbitmq.client.impl import DefaultCredentialsProvider
from jakarta.inject import Singleton
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener


@Singleton
class ConnectionFactoryInterceptor(BeanCreatedEventListener[ConnectionFactory]):

    def onCreated(self, event: BeanCreatedEvent[ConnectionFactory]) -> ConnectionFactory:
        connection_factory = event.getBean()
        connection_factory.setCredentialsProvider(DefaultCredentialsProvider("guest", "guest"))
        return connection_factory
